src/SAML2/XML/saml/AuthnContext.php

```diff
@@ -0,0 +1,266 @@
+    {
+        return $this->authnContextDeclRef;
+    }
+
+
+    /**
+     * Set the value of the authnContextDeclRef-property
+     *
+     * @param \SAML2\XML\saml\AuthnContextDeclRef|null $authnContextDeclRef
+     * @return void
+     */
+    public function setAuthnContextDeclRef(?AuthnContextDeclRef $authnContextDeclRef): void
+    {
+        $this->authnContextDeclRef = $authnContextDeclRef;
+    }
+
+
+    /**
+     * Collect the value of the authnContextDecl-property
+     *
+     * @return \SAML2\XML\saml\AuthnContextDecl|null
+     */
+    public function getAuthnContextDecl(): ?AuthnContextDecl
+    {
+        return $this->authnContextDecl;
+    }
+
+
+    /**
+     * Set the value of the authnContextDecl-property
+     *
+     * @param \SAML2\XML\saml\AuthnContextDecl|null $authnContextDecl
+     * @return void
+     */
+    public function setAuthnContextDecl(?AuthnContextDecl $authnContextDecl): void
+    {
+        $this->authnContextDecl = $authnContextDecl;
+    }
+
+
+    /**
+     * Collect the value of the authenticatingAuthorities-property
+     *
+     * @return \SAML2\XML\saml\AuthenticatingAuthority[]|null
+     */
+    public function getAuthticatingAuthorities(): ?array
+    {
+        return $this->authenticatingAuthorities;
+    }
+
+
+    /**
+     * Set the value of the authenticatingAuthorities-property
+     *
+     * @param \SAML2\XML\saml\AuthenticatingAuthority[]|null $authenticatingAuthorities
+     * @return void
+     */
+    public function setAuthenticatingAuthorities(?array $authenticatingAuthorities): void
+    {
+        $this->authenticatingAuthorities = $authenticatingAuthorities;
+    }
+
+
+    /**
+     * Convert XML into a AuthnContext
+     *
+     * @param \DOMElement $xml The XML element we should load
+     * @return \SAML2\XML\saml\AuthnContext
+     */
+    public static function fromXML(DOMElement $xml): object
+    {
+        /** @var \DOMElement[] $authnContextClassRef */
+        $authnContextClassRef = Utils::xpQuery($xml, './saml_assertion:AuthnContextClassRef');
+        Assert::maxCount($authnContextClassRef, 1);
+
+        /** @var \DOMElement[] $authnContextDeclRef */
+        $authnContextDeclRef = Utils::xpQuery($xml, './saml_assertion:AuthnContextDeclRef');
+        Assert::maxCount($authnContextDeclRef, 1);
+
+        /** @var \DOMElement[] $authnContextDecl */
+        $authnContextDecl = Utils::xpQuery($xml, './saml_assertion:AuthnContextDecl');
+        Assert::maxCount($authnContextDecl, 1);
+
+        if (!empty($authnContextClassRef)) {
+            Assert::oneOf(
+                [],
+                [$authnContextDecl, $authnContextDeclRef],
+                'Can only have one of AuthnContextDecl/AuthnContextDeclRef'
+            );
+        } else {
+            Assert::false(
+                empty($authnContextDecl) && empty($authnContextDeclRef),
+                'You need either an AuthnContextDecl or an AuthnContextDeclRef'
+            );
+        }
```

Comment:
We also have this exact check in the constructor. Can we skip it here, considering we're just passing the processed values to it later in the method?